python/SAGEMAKER_ENDPOINT_CONFIG_KMS_KEY_CONFIGURED/SAGEMAKER_ENDPOINT_CONFIG_KMS_KEY_CONFIGURED_test.py

```diff
@@ -0,0 +1,249 @@
+DEFAULT_RESOURCE_TYPE = 'AWS::::Account'
+
+#############
+# Main Code #
+#############
+
+CONFIG_CLIENT_MOCK = MagicMock()
+STS_CLIENT_MOCK = MagicMock()
+SAGEMAKER_CLIENT_MOCK = MagicMock()
+
+class Boto3Mock():
+    @staticmethod
+    def client(client_name, *args, **kwargs):
+        if client_name == 'config':
+            return CONFIG_CLIENT_MOCK
+        if client_name == 'sts':
+            return STS_CLIENT_MOCK
+        if client_name == 'sagemaker':
+            return SAGEMAKER_CLIENT_MOCK
+        raise Exception("Attempting to create an unknown client")
+
+sys.modules['boto3'] = Boto3Mock()
+
+RULE = __import__('SAGEMAKER_ENDPOINT_CONFIG_KMS_KEY_CONFIGURED')
+
+class ComplianceTest(unittest.TestCase):
+
+    rule_parameters_scenario3 = '{"keyIds":"arn:aws:kms:us-east-1:305333956852:key/ae25566a-c0d4-4ed2-8131-7c98e9487h3d, arn:aws:kms:us-east-1:305333956852:key/ae25566a-c0d4-4edg-8131-7c98e9487e3d"}'
+    list_endpoints_scenario3 = [{'EndpointConfigs':[{'EndpointConfigName':'endpoint1'}, {'EndpointConfigName':'endpoint2'}]}]
+    described_endpoints_scenario3 = [{'EndpointConfigName':'endpoint1', 'EndpointConfigArn': 'arn:aws:sagemaker:us-east-1:305333956852:endpoint-config/endpoint1'}, {'EndpointConfigName': 'endpoint2', 'EndpointConfigArn': 'arn:aws:sagemaker:us-east-1:305333956852:endpoint-config/endpoint2'}]
+
+    rule_parameters_scenario4 = '{"keyIds":"arn:aws:kms:us-east-1:305333956852:key/ae25566a-c0d4-4ed2-8131-7c98e9487h3d, arn:aws:kms:us-east-1:305333956852:key/ae25566a-c0d4-4edg-8131-7c98e9487e3d"}'
+    list_endpoints_scenario4 = [{'EndpointConfigs':[{'EndpointConfigName':'endpoint1'}, {'EndpointConfigName':'endpoint2'}]}]
+    described_endpoints_scenario4 = [{'EndpointConfigName':'endpoint1', 'KmsKeyId': 'arn:aws:kms:us-east-1:305333956852:key/ae25566a-c0d4-4ed2-8131-7c98e9487e3d', 'EndpointConfigArn': 'arn:aws:sagemaker:us-east-1:305333956852:endpoint-config/endpoint1'}, {'EndpointConfigName': 'endpoint2', 'EndpointConfigArn': 'arn:aws:sagemaker:us-east-1:305333956852:endpoint-config/endpoint2', 'KmsKeyId': 'arn:aws:kms:us-east-1:305333956852:key/ae25566a-c0d4-4ed2-8131-7c98e9487e3d'}]
+
+    rule_parameters_scenario5 = '{"keyIds":"arn:aws:kms:us-east-1:305333956852:key/ae25566a-c0d4-4ed2-8131-7c98e9487e3d, arn:aws:kms:us-east-1:305333956852:key/ae25566a-c0d4-4ed2-8131-7c98e9487e3f"}'
+    list_endpoints_scenario5 = [{'EndpointConfigs':[{'EndpointConfigName':'endpoint1'}, {'EndpointConfigName':'endpoint2'}]}]
+    described_endpoints_scenario5 = [{'EndpointConfigName':'endpoint1', 'KmsKeyId': 'arn:aws:kms:us-east-1:305333956852:key/ae25566a-c0d4-4ed2-8131-7c98e9487e3d', 'EndpointConfigArn': 'arn:aws:sagemaker:us-east-1:305333956852:endpoint-config/endpoint1'}, {'EndpointConfigName': 'endpoint2', 'EndpointConfigArn': 'arn:aws:sagemaker:us-east-1:305333956852:endpoint-config/endpoint2', 'KmsKeyId': 'arn:aws:kms:us-east-1:305333956852:key/ae25566a-c0d4-4ed2-8131-7c98e9487e3f'}]
+
```

Comment:
The same rule parameters and list_endpoints can be used across scenarios 3, 4 and 5. Just use modified described_endpoints to vary compliance scenarios.